wgpu-hal/src/gles/egl.rs

```diff
@@ -346,14 +349,34 @@ impl AdapterContext {
     }
 }
 
+impl Drop for AdapterContext {
+    fn drop(&mut self) {
+        // Context must be current when dropped. See safety docs on
+        // `glow::HasContext`.
+        //
+        // NOTE: This is only set to `None` by `Adapter::new_external` which
+        // requires the context to be current when anything that may be holding
+        // the `Arc<AdapterShared>` is dropped.
+        self.egl.as_ref().map(|egl| {
+            egl.make_current();
+        });
+        let glow = self.glow.get_mut();
+        // SAFETY: Field not used after this.
+        unsafe { ManuallyDrop::drop(glow) };
+        self.egl.as_ref().map(|egl| {
+            egl.unmake_current();
```

Comment:
I just realized that if the drop panics for some reason then we won't call `unmake_current`, might want to adjust this to use a guard to call `unmake_current`?